portfolio/src/main/webapp/script.js

```diff
@@ -58,28 +58,105 @@ function presentFeedbackReceipt() {
  */
 function executeConsoleCode() {
   // Obtain user input JavaScript code.
-  const promptString = "Please enter your JavaScript code:\n";
+  const promptString = 'Please enter your JavaScript code:\n';
   const jsCode = window.prompt(promptString);
   if (!jsCode) {
-    window.alert("No code is received.");
+    window.alert('No code is received.');
     return;
   }
 
   // Execute JavaScript code; catch and return error if error occurs.
   let executionResult = null;
   try {
     executionResult = Function('"use strict"; return ' + jsCode)();
-  }
-  catch (err) {
-    window.alert(`An error has occurred:\n`
-                 + `> ${jsCode}\n`
-                 + `< ${err}\n`);
+  } catch (err) {
+    window.alert(`An error has occurred:\n` +
+                 `> ${jsCode}\n` +
+                 `< ${err}\n`);
     return;
   }
 
   // Present execution results in a pop-up window.
-  const consoleOutput = `CONSOLE\n`
-                        + `> ${jsCode}\n`
-                        + `< ${executionResult}\n`;
+  const consoleOutput = `CONSOLE\n` +
+                        `> ${jsCode}\n` +
+                        `< ${executionResult}\n`;
   window.alert(consoleOutput);
 }
+
+/**
+ * Search for and display top 3 Wikipedia matches without reloading the page.
+ */
+function searchWikipedia() {
+  helperSearchWikipedia()
+    .then(wikipediaJSON => {
+      // Display Wikipedia information on the page.
+      const wikipediaContainer = document.getElementById('wikipedia-container');
+      const wikipediaSearchResult = 'Top 3 Search Results:\n' +
+                                    wikipediaJSON[1].join("\n");
+      wikipediaContainer.innerText = wikipediaSearchResult;
```

Comment:
Does the wikipediaJSON have the url information as well? It'd be cool to make these results clickable to go to the actual wikipedia pages. No worries if this is too much trouble though (or feel free to do this in a different pull request if you're interested!)

Reply:
Yes it does have the url information actually! I can try adding HTML anchor elements